analyses/malawi/flooding_trigger/02_floodscan_events.md

```diff
@@ -0,0 +1,169 @@
+PRIVATE_DIR = config.DATA_PRIVATE_DIR
+EXPLORE_DIR = PRIVATE_DIR / 'exploration' / 'mwi' / 'flooding'
+
+SAVE_PLOT = False
+SAVE_DATA = False
+
+stations_adm2 = {
+    'G1724': 'Nsanje',
+    'G2001': 'Chikwawa'
+}
+```
+
+Take a look at an example of the Floodscan data to get a general sense of the scale. Here we're clipping it to a bounding box around the two districts of interest.
+
+```python
+fs = floodscan.Floodscan()
+fs_raw = fs.read_raw_dataset()
+
+ds_sel = (
+    fs_raw.sel(time='2015-02-10')[['SFED_AREA']]
+    .to_array()
+    .rio.write_crs("EPSG:4326")
+    .rio.set_spatial_dims(x_dim="lon", y_dim="lat")
+)
+
+geometries = [
+    {
+        'type': 'Polygon',
+        'coordinates': [
+            [[34.1011082082,-17.12960312],
+             [35.4722024947,-17.12960312],
+             [35.4722024947,-15.5314191359],
+             [34.1011082082,-15.5314191359],
+             [34.1011082082,-17.12960312]
+            ]]
+    }
+]
+clipped = ds_sel.rio.clip(geometries)
+clipped.plot()
+if SAVE_PLOT: plt.savefig(PLOT_DIR / 'floodscan_overview.png')
+```
+
+Read in the pre-processed Floodscan data and filter for our variables and districts of interest.
+
+```python
+df_floodscan_path = Path(os.environ['AA_DATA_DIR'])/'private'/'processed'/'mwi'/'floodscan'/'mwi_floodscan_stats_adm2.csv'
+df_floodscan = pd.read_csv(df_floodscan_path)
+df_floodscan = df_floodscan[['ADM2_EN','date', 'mean_cell', 'max_cell', 'min_cell']]
+df_floodscan = df_floodscan[df_floodscan['ADM2_EN'].isin(stations_adm2.values())]
+df_floodscan['date'] = pd.to_datetime(df_floodscan['date'])
+```
+
+Calculate the 5-day rolling average to smooth out potential noise and create simple plots to show the changes in flooding fraction over time.
+
+```python
+for station in stations_adm2.values():
+    fig, ax = plt.subplots()
+    df_floodscan_sel = df_floodscan[df_floodscan['ADM2_EN']==station]
+    df_floodscan_sel['mean_cell_rolling'] = df_floodscan_sel['mean_cell'].transform(lambda x: x.rolling(5, 1).mean())
+    sns.lineplot(data=df_floodscan_sel, x="date", y="mean_cell", lw=0.25, label='Original')
+    sns.lineplot(data=df_floodscan_sel, x="date", y="mean_cell_rolling", lw=0.25, label='5-day moving\navg')   
+    ax.set_ylabel('Mean flooded fraction')
+    ax.set_xlabel('Date')
+    ax.set_title(f'Flooding in {station}, 1998-2020')
+    ax.legend()
+    if SAVE_PLOT: plt.savefig(PLOT_DIR / f'{station}_flooding_fraction.png')
+```
+
+Remove dates outside of the rainy season. We're loosely defining the rainy season as being between October - April (inclusive). 
+
+```python
+df_floodscan['month'] = pd.DatetimeIndex(df_floodscan['date']).month
+df_floodscan_rainy = df_floodscan.loc[(df_floodscan['month'] >= 10) | (df_floodscan['month'] <= 4)]
+```
+
+Now with this cleaned up data we can identify consecutive dates of significantly above average (>3 standard deviations) surface water coverage. We'll consider these to be flood events. 
```

Comment:
Is there a reason you chose 3 standard deviations? Also if not, would be good to add that this is based on experience/gut feeling. Would it also make sense to make this 3 a variable?